pkg/autoscaler/autoscaler/autoscaler_manager.go

```diff
@@ -55,52 +68,85 @@ func (am *autoScalerManager) Sync(tac *v1alpha1.TidbClusterAutoScaler) error {
 		tac.Spec.Cluster.Namespace = tac.Namespace
 	}
 
-	tcNamespace := tac.Spec.Cluster.Namespace
-	tc, err := am.tcLister.TidbClusters(tcNamespace).Get(tcName)
+	tc, err := am.cli.PingcapV1alpha1().TidbClusters(tac.Spec.Cluster.Namespace).Get(tcName, metav1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Target TidbCluster Ref is deleted, empty the auto-scaling status
 			resetAutoScalingAnn(tac)
+			return nil
 		}
 		return err
 	}
 	checkAndUpdateTacAnn(tac)
-	oldTCSpec := tc.Spec.DeepCopy()
+	oldTc := tc.DeepCopy()
 	if err := am.syncAutoScaling(tc, tac); err != nil {
 		return err
 	}
-	if err := am.syncTidbClusterReplicas(tc, oldTCSpec); err != nil {
+	if err := am.syncTidbClusterReplicas(tc, oldTc); err != nil {
 		return err
 	}
-	return am.syncAutoScalingStatus(tc, oldTCSpec, tac)
+	return am.syncAutoScalingStatus(tc, oldTc, tac)
 }
 
 func (am *autoScalerManager) syncAutoScaling(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler) error {
 	if tac.Spec.MetricsUrl == nil {
 		return fmt.Errorf("tidbclusterAutoScaler[%s/%s]' metrics url should be defined explicitly", tac.Namespace, tac.Name)
 	}
-	client, err := promClient.NewClient(promClient.Config{Address: *tac.Spec.MetricsUrl})
+	c, err := promClient.NewClient(promClient.Config{Address: *tac.Spec.MetricsUrl})
 	if err != nil {
 		return err
 	}
 	defaultTAC(tac)
-	if err := am.syncTiKV(tc, tac, client); err != nil {
-		return err
+	if err := am.syncTiKV(tc, tac, c); err != nil {
+		klog.Errorf("tac[%s/%s] tikv sync failed, continue to sync next, err:%v", tac.Namespace, tac.Name, err)
 	}
-	if err := am.syncTiDB(tc, tac, client); err != nil {
-		return err
+	if err := am.syncTiDB(tc, tac, c); err != nil {
+		klog.Errorf("tac[%s/%s] tidb sync failed, continue to sync next, err:%v", tac.Namespace, tac.Name, err)
```

Comment:
Still have to check if any error occurred, should return error in the end to re-queue.

Reply:
The caller function also has to handle this.